src/core/core.controller.js

```diff
@@ -129,6 +129,7 @@ function updateConfig(chart) {
 		newOptions);
 
 	chart.options = chart.config.options = newOptions;
+	chart._animation = newOptions.animation && newOptions.animation.duration;
```

Comment:
I think you also need to consider the other types of animations or those animations will break: https://www.chartjs.org/docs/latest/general/performance.html#disable-animations